pkg/util/generic.go

```diff
@@ -82,6 +83,35 @@ func ProtoClone[T proto.Message](msg T) T {
 	return proto.Clone(msg).(T)
 }
 
+func NewMessage[T proto.Message]() T {
```

Comment:
Yeah we definitely should re-organize our util packages. I don't want to do that here though since it would cause a lot of conflicts elsewhere. but once all this is merged we should do a bit of tidying up

Reply:
Can we create a backlog issue so we don't forget this

Reply:
https://github.com/rancher/opni/issues/1702